client/components/pulsing-dot/index.jsx

```diff
@@ -12,6 +17,30 @@ var PulsingDot = React.createClass( {
 		};
 	},
 
+	_loadingStatTimeout: null,
+
+	componentWillUnmount: function() {
+		if ( this._loadingStatTimeout ) {
+			clearTimeout( this._loadingStatTimeout );
+			this._loadingStatTimeout = null;
+		}
+	},
+
+	componentWillReceiveProps: function( nextProps ) {
+		if ( nextProps.active === this.props.active ) {
+			return;
+		}
+
+		if ( nextProps.active ) {
+			this._loadingStatTimeout = setTimeout( () => {
+				analytics.mc.bumpStat( 'calypso_chunk_waiting', this.props.chunkName );
```

Comment:
Checking with @apeatling it seems we do underscores which are then converted to spaces for display?